src/ngraph/pass/constant_folding.cpp

```diff
@@ -763,3 +766,216 @@ void pass::ConstantFolding::construct_constant_quantize()
     this->add_matcher(
         quantize_matcher, constant_quantize_callback, PassProperty::REQUIRE_STATIC_SHAPE);
 }
+
+template <typename TI, typename TO>
+shared_ptr<op::Constant> fold_constant_convert_helper1(shared_ptr<op::Constant> constant,
+                                                       const element::Type& output_element_type)
+{
+    auto out_shape = constant->get_shape();
+    vector<TO> out_vec(shape_size(out_shape));
+
+    runtime::reference::convert<TI, TO>(
+        constant->get_vector<TI>().data(), out_vec.data(), shape_size(out_shape));
+
+    return make_shared<op::Constant>(output_element_type, out_shape, out_vec);
+}
+
+template <typename TI>
+shared_ptr<op::Constant> fold_constant_convert_helper0(shared_ptr<op::Constant> constant,
+                                                       const element::Type& output_element_type)
+{
+    if (output_element_type == element::boolean)
```

Comment:
Use a switch statement here and add the #pragma stuff to make it a compile-time error if an enum is missing from the list.